sdk/ai/ai-projects/samples-dev/agents/code_interpreter.ts

```diff
@@ -0,0 +1,99 @@
+
+const client = AIProjectsClient.fromConnectionString(connectionString || "", new DefaultAzureCredential());
+
+// Upload file and wait for it to be processed
+const localFileStream = fs.createReadStream("nifty_500_quarterly_results.csv");
+const localFile = await client.agents.uploadFile(localFileStream, "assistants", "my-local-file");
+
+console.log(`Uploaded local file, file ID : ${localFile.id}`);
+
+// Notice that CodeInterpreter must be enabled in the agent creation, otherwise the agent will not be able to see the file attachment
+const agent = await client.agents.createAgent("gpt-4o-mini", {
+  name: "my-agent",
+  instructions: "You are a helpful agent",
+  tools: [{type: "code_interpreter"} as CodeInterpreterToolDefinition],
+  tool_resources: { code_interpreter: {file_ids: [localFile.id]} } as ToolResources
+});
+console.log(`Created agent, agent ID: ${agent.id}`);
+
+// Create a thread
+const thread = await client.agents.createThread();
+console.log(`Created thread, thread ID: ${thread.id}`);
+
+// Create a message
+const message = await client.agents.createMessage(thread.id, { 
+  role: "user", 
+  content: "Could you please create a bar chart in the TRANSPORTATION sector for the operating profit from the uploaded CSV file and provide the file to me?" 
+});
+
+console.log(`Created message, message ID: ${message.id}`);
+
+// Create and execute a run
+let run = await client.agents.createRun(thread.id, agent.id);
+while (run.status === "queued" || run.status === "in_progress") {
+  await delay(1000);
+  run = await client.agents.getRun(thread.id, run.id);
+}
+if (run.status === "failed") {
+    // Check if you got "Rate limit is exceeded.", then you want to get more quota
+    console.log(`Run failed: ${run.last_error}`);
+}
+console.log(`Run finished with status: ${run.status}`);
+
+// Delete the original file from the agent to free up space (note: this does not delete your version of the file)
+await client.agents.deleteFile(localFile.id);
+console.log("Deleted file");
+
+// Print the messages from the agent
+const messages = await client.agents.listMessages(thread.id);
+console.log("Messages:", messages);
+
+// Get most recent message from the assistant
+const assistantMessage = messages.data.find(msg => msg.role === "assistant");
+if (assistantMessage) {
+  const textContent = assistantMessage.content.find(content => isOutputOfType<MessageTextContentOutput>(content, "text")) as MessageTextContentOutput;
+  if (textContent) {
+    console.log(`Last message: ${textContent.text.value}`);
+  }
+}
+
+// Save the newly created file
+messages.data.forEach( m => {
+  m.content.forEach(async (content: MessageContentOutput) => {
+      if (isOutputOfType<MessageImageFileContentOutput>(content, "image_file")) {
+          const imageContent = content as MessageImageFileContentOutput;
+          const file = await client.agents.uploadFile(fs.createReadStream(imageContent.image_file.file_id), "assistants", "uploaded-image-file");
```

Comment:
Curios why is it needed to update file?

Reply:
@ganeshyb I was trying to pull directly from the message with the image file to save it but the loop was unnecessary, I removed it. Right now the file is saved but it is empty. I believe for the file content to be added I need to use the getFileContent method but there is still an issue with the stream, do you know of a fix?

Reply:
This is fixed in the new changes to getFileContent, image is now able to be generated properly.